### PR TITLE
Add resource support to logger

### PR DESCRIPTION
**Pull request by reyang**
Fixes #.

## Changes

* Added `Resource` support to OpenTelemetryLoggerProvider.
* Added `Resource` support to the console exporter.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs
Comment (cijothomas) [inline src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:52]: Could we make OpenTelemetryLoggerProvider implement `BaseProvider`, (similar to `TracerProviderSdk`) and then modify `ProviderExtensions.GetResource` to return the resource?

Comment (cijothomas) [reply src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:52]: This can avoid storing Resource inside LogRecord, but still make it available for Processor/Exporter.

Comment (reyang) [reply src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs:52]: I've updated the PR, please take another look.

File: src/OpenTelemetry/.publicApi/netstandard2.0/PublicAPI.Shipped.txt
Comment (reyang) [inline src/OpenTelemetry/.publicApi/netstandard2.0/PublicAPI.Shipped.txt:175]: This is a bit messy - changing Dispose from virtual to override is causing the complaint from the tooling.

Comment (cijothomas) [reply src/OpenTelemetry/.publicApi/netstandard2.0/PublicAPI.Shipped.txt:175]: As long as this is not considered an actual breaking change, we should be fine.

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
@open-telemetry/dotnet-approvers Could use an additional review before merge!

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Pull request merged**
